packages/asset-kit-block/src/components/AssetSelection.tsx

```diff
@@ -24,17 +24,16 @@ export const AssetSelection = ({
 
     const onOpenAssetChooser = () => {
         openAssetChooser(
-            (assetsObject) => {
+            async (assetsObject) => {
                 setIsUploadingAssets(true);
                 const assetsIds = Array.from(assetsObject).map((asset) => asset.id);
                 saveDownloadUrl('');
-                addAssetIdsToKey(ASSET_SETTINGS_ID, assetsIds).then(() => setIsUploadingAssets(false));
+                await addAssetIdsToKey(ASSET_SETTINGS_ID, assetsIds).then(() => setIsUploadingAssets(false));
```

Comment:
Will this keep the asset chooser open until the API request is done? What was the  issue with showing the loading indicator?

Reply:
Reverted, as it was highly useless in combination with `.then`